hive/src/main/java/org/apache/iceberg/hive/HiveCatalogs.java

```diff
@@ -47,6 +53,13 @@ public static HiveCatalog loadCatalog(Configuration conf) {
     return CATALOG_CACHE.get(metastoreUri, uri -> new HiveCatalog(conf));
   }
 
+  public static HiveCatalog loadCatalogWithLegacyReadFallback(Configuration conf) {
```

Comment:
I guess we need to call this from IcebergSource to trigger the fallback logic?

Reply:
Correct. We can decide whats the best way to expose this. Maybe we can also decide whether to fallback or not based on a config variable.